includes/data-stores/class-wcs-orders-table-subscription-data-store.php

```diff
@@ -603,4 +603,53 @@ public function get_subscription_customer_ids() {
 
 		return $wpdb->get_col( "SELECT DISTINCT customer_id FROM {$table_name} WHERE type = 'shop_subscription'" ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
 	}
+
+	/**
+	 * Deletes all rows in the postmeta table with the given meta key.
+	 *
+	 * @param string $meta_key The meta key to delete.
+	 */
+	public function delete_all_metadata_by_key( $meta_key ) {
+		global $wpdb;
+
+		$wpdb->delete( self::get_meta_table_name(), [ 'meta_key' => $meta_key ], [ '%s' ] ); // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+	}
+
+	/**
+	 * Fetches meta data directly from the wp_wc_orders_meta table.
+	 *
+	 * @param int    $subscription_id The subscription ID.
+	 * @param string $meta_key        The meta key to fetch.
+	 * @param bool   $single          Whether to return a single value.
+	 *
+	 * @return mixed The meta value.
+	 */
+	public function get_metadata_by_key( $subscription_id, $meta_key, $single = true ) {
```

Comment:
I'm surprised that Woo core doesn't provide a helper/abstraction for this.

Reply:
They have methods for fetching meta data by key from the order or subscription in memory but because of a quirk with this class (the meta keys are ignored when being read from the database) we needed a way to fetch the data via the key directly from the database. 

In the end I was able to find a way around it in [372af8c](https://github.com/Automattic/woocommerce-subscriptions-core/pull/256/commits/372af8c0d310f0655341f6f551903d2bfeaf5e36) and deleted this function in [7bfa7e2](https://github.com/Automattic/woocommerce-subscriptions-core/pull/256/commits/7bfa7e2b787e3debefbfa31f0691a9c36774ab4c)

Reply:
Nice! If the Woo apis should be improved maybe we can log that request with woo core.